internal/docker/docker.go

```diff
@@ -190,3 +190,20 @@ func Copy(containerName, containerPath, localPath string) error {
 	}
 	return nil
 }
+
+// CheckDaemonRunning checks if the Docker daemon is running.
+// Returns nil if the daemon is running, or an error if it's not.
+func CheckDaemonRunning() error {
+	cmd := exec.Command("docker", "info")
```

Comment:
This commands has a large output. It should not be an issue, but maybe it could be format the output to just shown a specific value (e.g. `ServerVersion`). 

```suggestion
	cmd := exec.Command("docker", "info", "--format", ""{{ .ServerVersion }}")
```